### PR TITLE
Leverage conditional requests to save API quotas

### LINKED ISSUE #23: Consider using conditional requests to check for changes
The GitHub API allows for conditional requests that do not count towards rate-limiting.

It could be interesting to leverage these and avoid making requests if they are not necessary.

https://docs.github.com/en/rest/overview/resources-in-the-rest-api#conditional-requests

### PR DESCRIPTION
**Pull request by ericboucher**
Fixes #23 

Add an "If-modified-since" header to try and leverage conditional requests from GitHub

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
> is it reasonable to add a context-aware header in `get_url_params`? I couldn't really find a better way to do this but let me know if you have other ideas.

@ericboucher This is currently the best (and only!) way to pass values from context to headers. It does make sense to have better support in the SDK, though.

File: tap_github/repository_streams.py [hunk 1]
```diff
@@ -2022,7 +2022,7 @@ def http_headers(self) -> dict:
 
         Mock a web browser user-agent.
         """
-        return {"User-agent": "Mozilla/5.0"}
+        return {"User-Agent": "Mozilla/5.0"}
```
Comment (laurentS) [inline tap_github/repository_streams.py:2025]: Slightly off-topic for this PR, but GitHub [request](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#user-agent-required) that a clear app name be used, should this say `tap-github` instead?

**Review by laurentS — COMMENTED** [review]
When you use the conditional requests header, the server might return an HTTP 304 status. Do we need to handle it in a particular way in the stream?

File: tap_github/repository_streams.py (hunk 1 above)
Comment (ericboucher) [reply tap_github/repository_streams.py:2025]: This is mocking user http requests, not pinging the API so I think we actually don't want to give a clear app name 😄

Comment (laurentS) [reply tap_github/repository_streams.py:2025]: Hmm, I thought this was pretending to be a web browser, but the stream is actually pinging the API, no? :thinking:

Comment (ericboucher) [reply tap_github/repository_streams.py:2025]: No, this stream is a bit convoluted but it is actually pinging GitHub's website. The switch happens in `scrape_dependents` and we use beautifulsoup to parse the html

Comment (laurentS) [reply tap_github/repository_streams.py:2028]: Ok, I totally missed that. Just added a short comment to make it more obvious.

**Review by laurentS — APPROVED** [review]

**Pull request merged**